tests/test_dpo_trainer.py

```diff
@@ -589,59 +589,6 @@ def test_dpo_trainer_generate_during_eval_no_wandb(self):
                 eval_dataset=dummy_dataset["test"],
             )
 
-    @require_peft
-    def test_dpo_lora_save(self):
-        from peft import LoraConfig, get_peft_model
-
-        lora_config = LoraConfig(
-            r=16,
-            lora_alpha=32,
-            lora_dropout=0.05,
-            bias="none",
-            task_type="CAUSAL_LM",
-        )
-
-        # lora model
-        model = AutoModelForCausalLM.from_pretrained(self.model_id)
-        model_peft = get_peft_model(model, lora_config)
-
-        training_args = DPOConfig(
-            output_dir=self.tmp_dir,
-            per_device_train_batch_size=2,
-            max_steps=3,
-            remove_unused_columns=False,
-            gradient_accumulation_steps=4,
-            learning_rate=9e-1,
-            eval_strategy="steps",
-            beta=0.1,
-            precompute_ref_log_probs=True,
-            report_to="none",
-        )
-
-        dummy_dataset = load_dataset("trl-internal-testing/zen", "standard_preference")
-
-        # dpo train lora model with a lora config
-        trainer = DPOTrainer(
-            model=model_peft,
-            ref_model=None,
-            args=training_args,
-            processing_class=self.tokenizer,
-            train_dataset=dummy_dataset["train"],
-            eval_dataset=dummy_dataset["test"],
-            peft_config=lora_config,
-        )
-
-        # train the model
-        trainer.train()
-
-        # save peft adapter
-        trainer.save_model()
-
-        try:
-            AutoModelForCausalLM.from_pretrained(self.tmp_dir)
-        except OSError:
-            pytest.fail("Loading the saved peft adapter failed")
```

Comment:
The reason why this test was passing is actually quite subtle:

It's because 

1. we use the same config for the base adapter and for the trained adapter and
2. when you call (line 606)
    ```python
    model_peft = get_peft_model(model, lora_config)
    ```
    it modifies `lora_config` object and adds `base_model_name_or_path`, see [here](https://github.com/huggingface/peft/blob/89a75aa12a143f82132195b814026bd1aa1a165e/src/peft/mapping_func.py#L66).

So when you "re-apply" the same LoRA config, it mistakenly uses the base model path, while it should be `None` (since the base model actually doesn't exists in this case).